src/Features/Core/Portable/ChangeSignature/Parameter.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+#nullable enable
+
+namespace Microsoft.CodeAnalysis.ChangeSignature
+{
+    internal abstract class Parameter
+    {
+        public abstract bool HasExplicitDefaultValue { get; }
+        public abstract string Name { get; }
+    }
+
+    internal sealed class ExistingParameter : Parameter
+    {
+        public IParameterSymbol Symbol { get; }
+
+        public ExistingParameter(IParameterSymbol param)
+        {
+            Symbol = param;
+        }
+
+        public override bool HasExplicitDefaultValue => Symbol.HasExplicitDefaultValue;
+        public override string Name => Symbol.Name;
+    }
+
+    internal sealed class AddedParameter : Parameter
+    {
+        public AddedParameter(ITypeSymbol type, string typeNameDisplayWithErrorIndicator, string parameter, string callSiteValue)
+        {
+            Type = type;
+            TypeNameDisplayWithErrorIndicator = typeNameDisplayWithErrorIndicator;
+            ParameterName = parameter;
+            CallSiteValue = callSiteValue;
+        }
+
+        public ITypeSymbol Type { get; set; }
+        public string ParameterName { get; set; }
+        public string CallSiteValue { get; set; }
```

Comment:
note: CodeGenSymbols can 'Attach' additional information to them to allow them to pass along data.  i think we likely can remove this indirection and just use IParameterSymbols.

Reply:
@CyrusNajmabadi Attach how?

Reply:
Filed as https://github.com/dotnet/roslyn/issues/42687